### PR TITLE
Feat: add test case documentation

### PR DESCRIPTION
**Pull request by Bai-Li-NOAA**
*What is the feature?*
This PR addresses issue from NOAA-FIMS/collaborative_workflow#65. It adds milestone 1 test cases documentation under the tests/ folder.

*How have you implemented the solution?*
- added milestone1_test_cases.md template to the repo
- added milestone1_test_cases.md to .gitbuildignore file 
- also renamed the test-place-holder.R without changing anything inside the file

*Does it impact any other area of the project?*
No.

*Does this change impact the model input or output?*
No.

*How to test this change*
All the GH Actions workflow passed except the R cmd check. The failed run was not caused by the changes in this PR.

### CODE REVIEW

File: tests/milestone1_test_cases.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## General test case documentation
+
+The test case documentation below is a general case to apply to many functions/modules. For individual functions/modules, please make detailed test cases for specific options, noting "same as the general test case" where appropriate.
+
+- Test ID: General test case
+- Feature to be tested
+    - The function/module returns correct output values given different input values
+    - The function/module returns error messages when users give wrong types of inputs
+    - The function/module notifies an error if the input value is outside the bound of the input parameter
```
Comment (ChristineStawitz-NOAA) [inline tests/milestone1_test_cases.md:11]: should be "notifies with an error"

File: tests/milestone1_test_cases.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+- Approach
+    - Prepare expected true values using R
+    - Run tests in R using testthat and compare output values with expected values
+    - Push tests to the working repository and run tests using GitHub Actions
+    - Submit pull request for code review
+
+- Evaluation criteria
+    - The tests pass if the output values equal to the expected true values
```
Comment (ChristineStawitz-NOAA) [inline tests/milestone1_test_cases.md:20]: You might want to make it clear to compare within a tolerance such that the values don't have to be exactly equal.

**Review by ChristineStawitz-NOAA — APPROVED** [review]
Looks good - just two small optional suggestions

File: tests/milestone1_test_cases.md (hunk 1 above)
Comment (Bai-Li-NOAA) [reply tests/milestone1_test_cases.md]: thanks for the fix and I revised the file.

File: tests/milestone1_test_cases.md (hunk 2 above)
Comment (Bai-Li-NOAA) [reply tests/milestone1_test_cases.md]: Thanks for the suggestion. I added some clarifications to the evaluation criteria.

**Pull request merged**